ozaria/site/components/play/PageIntroLevel/index.vue

```diff
@@ -74,6 +77,13 @@
       {
         loadIntroLevel: async function () {
           this.dataLoaded = false
+
+          // Reading query params because this is rendered via backbone router and cannot be directly passed in as props
+          // They need to be in a specific order in the url to read and send them as props directy from backbone router, hence using query params here.
+          this.courseInstanceId = this.courseInstanceId || utils.getQueryVariable('course-instance')
+          this.codeLanguage = this.codeLanguage || utils.getQueryVariable('code-language')
+          this.courseId = this.courseId || utils.getQueryVariable('course')
+          this.campaignId = this.campaignId || utils.getQueryVariable('campaign')
           try {
```

Comment:
Because we are depending on the query params, what happens if a user messes with them? Should we add the
`TODO handle_error_ozaria` if the error is caught? (I know there is already a comment there. But just to follow our new conventions).

This might be a critical page that may need to handle invalid user input in some way (like maybe returning them to a unit map for example).

Reply:
Yeah will update the comment in the catch block. Agree that user can mess with them irrespective of what router we use.